### PR TITLE
[WIP] [Help requested] Basic groundwork for markings feature

### PR DESCRIPTION
**Pull request by Poojawa**
Basic ground work.

Using Paradise's body marking system they've implimented. 

/datum/preferences
var/r_markings = 0
 var/g_markings = 0
 var/b_markings = 0
 var/m_style = "None"

Does not actually allow for any vars to be used in 01_ears.dm. I'll make a body markings setup if it's needed though.

https://gyazo.com/ee412e991493b22de693c22a34f977d5

Too tired to bother trying to figure it out

### CODE REVIEW

**Comment by JoanRisu** [comment]
This failed the check so something is amiss somewhere or the checker isn't foolproof. I dunno. Coders check and make sure.

**Comment by Arokha** [comment]
Doesn't compile. Just look at the details button when it fails and it tells you what.

**Comment by SpadesNeil** [comment]
This is a big change so DNM until tested (once Travis issues are resolved that is).

**Comment by Poojawa** [comment]
I.. i know it doesn't compile yet, but I've made this to the working markings branch so people can look at it also. The gyazo link shows a big chuck on why it doesn't

**Comment by Arokha** [comment]
Fuggin' skittles labels up in here. Just leave it with DNM while it's WIP.

**Comment by SpadesNeil** [comment]
Shut up I like my skittles.

**Comment by Poojawa** [comment]
Aaaaaa. really don't know how @Leshana did code hook magics for ears and tails, and that's what's holding me up majorly because I am unsure at what needs to be defined and where.

**Comment by Poojawa** [comment]
AAAAAAAAAAAAAAAAA what even

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -107,6 +107,7 @@ Please contact me on #coderbus IRC. ~Carn x
 
 //Human Overlays Indexes/////////
 #define MUTATIONS_LAYER			1
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: The hell is this?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 2]
```diff
@@ -134,6 +135,34 @@ Please contact me on #coderbus IRC. ~Carn x
 #define FIRE_LAYER				26		//If you're on fire
 #define TARGETED_LAYER			27		//BS12: Layer for the target overlay from weapon targeting system
 #define TOTAL_LAYERS			27
+=======
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: And this? The hell is this?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 3]
```diff
@@ -134,6 +135,34 @@ Please contact me on #coderbus IRC. ~Carn x
...
+#define HEAD_LAYER				19
+#define COLLAR_LAYER			20
+#define HANDCUFF_LAYER			21
+#define LEGCUFF_LAYER			22
+#define L_HAND_LAYER			23
+#define R_HAND_LAYER			24
+#define FIRE_LAYER				25		//If you're on fire
+#define TARGETED_LAYER			26		//BS12: Layer for the target overlay from weapon targeting system
+#define TOTAL_LAYERS			26
+>>>>>>> origin/bodymarkings
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: The hell are these?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 4]
```diff
@@ -330,6 +360,7 @@ var/global/list/damage_icon_parts = list()
 	//tail
 	update_tail_showing(0)
 
+<<<<<<< HEAD
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: The hell are those?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 5]
```diff
@@ -339,6 +370,38 @@ var/global/list/damage_icon_parts = list()
 		for(var/category in all_underwear)
 			var/datum/category_item/underwear/UWI = all_underwear[category]
 			overlays_standing[UNDERWEAR_LAYER] += UWI.generate_image(all_underwear_metadata[category])
+=======
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Why are these in your code?

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 6]
```diff
@@ -339,6 +370,38 @@ var/global/list/damage_icon_parts = list()
...
+				return //If the head is destroyed and it is the organ the marking is located on, get us out of here. This prevents floating optical markings on decapitated IPCs, for example.
+			var/icon/markings_s = new/icon("icon" = marking_style.icon, "icon_state" = "[marking_style.icon_state]_s")
+			if(marking_style.do_colouration)
+				markings_s.Blend(rgb(r_markings, g_markings, b_markings), ICON_ADD)
+			markings_standing.Blend(markings_s, ICON_OVERLAY)
+		else
+			//warning("Invalid m_style for [species.name]: [m_style]")
+
+	overlays_standing[MARKINGS_LAYER]	= image(markings_standing)
+>>>>>>> origin/bodymarkings
```
Comment (SpadesNeil) [inline code/modules/mob/living/carbon/human/update_icons.dm]: No wonder this won't compile.

**Comment by SpadesNeil** [comment]
Also why is this 400 file changes? This should NOT be 400 file changes.

**Comment by JoanRisu** [comment]
It's going into an outdated branch from May 23rd instead of the master.

**Comment by SpadesNeil** [comment]
I'm going to close this by the end of the week if not brought up to date.

**Pull request closed without merging**